### ISSUE TITLE
Internal server error trying to access an existing issue

### ISSUE THREAD

**Issue by strk** [go-gitea/gitea#10384]
See what happens here: https://try.gitea.io/cezar97/Gitea_Issue_Logout/issues/4
Gitea version (or commit ref): 1.12.0+dev-345-gefbd7ca39

**Comment by guillep2k** [comment]
I can access the issue no problem. 🤔

**Comment by lunny** [comment]
It maybe a random problem.

**Comment by strk** [comment]
I can access it from an anonymous window, not when I'm logged in.
Can you access it when logged in ?
Do you have access to logs to spot what's going on ? I think log access should be available to all maintainers or at least owners of the project, to make the `try` service more useful.

**Comment by guillep2k** [comment]
I can access both logged in and anonymous.

Regarding the access to logs, AFAIU they're kept strictly private.

**Comment by strk** [comment]
On Fri, Feb 21, 2020 at 04:48:36AM -0800, guillep2k wrote:
> I can access both logged in and anonymous.

Then it must be related to my user configuration ?
I get 100% reproducibility

> Regarding the access to logs, AFAIU they're kept strictly private.

Private to whom ? Owners must have access, or what's the use of it ?

**Comment by guillep2k** [comment]
There was one user that had problems today because of their cached js content. Can you try again with a clean cache, just in case?

**Comment by mcmillen** [comment]
I am seeing this issue as well:

https://git.semicolin.games/semicolin/sneak/issues/4

Gets a 500 error for me every time, logged in or no. I've tried clearing cache, using an incognito window, etc.

There's a useful error line in the logs from trying to load this issue:

`2020/02/25 22:06:32 ...outers/repo/issue.go:882:ViewIssue() [E] LoadDepIssueDetails: issue does not exist [id: 25, repo_id: 0, index: 0]`

A rough guess based on this error message: maybe there's a false dependency between my issue 4 and an issue 25 that doesn't exist? (My repo only has 23 issues.)

**Comment by zeripath** [comment]
The log is giving you the issue id not the index of the issue. That log is suggesting that somehow your issue indexed 4 has gained a dependency on issue with id 25 in your db. If I had to guess that would be an issue in a deleted repo?

**Comment by mcmillen** [comment]
Could be! I have deleted one repo; I don't *think* it had any Issues associated with it, but I could be wrong. I'm not sure how to check for sure.

In the mean time I have "fixed" the issue by closing it via a commit comment ("fixes #4"), so that at least the broken issue doesn't show up in my list of open issues.

**Comment by lunny** [comment]
@mcmillen what's version of your gitea instance?

**Comment by mcmillen** [comment]
Gitea 1.11.1.

I verified that the issue also reproduces on 1.11.0.

**Comment by lunny** [comment]
This hasn't been resolved totally. I also need to allow deleted dependencies issues.

**Comment by lunny** [comment]
Since we cannot add migration to v1.11, we can add a migration on v1.12 and a command line subcommand to fix it.

**Comment by guillep2k** [comment]
Users that need a fix for this while still running 1.11 and can't upgrade to 1.12 (not released at this moment), can run the following in their databases:

```
DELETE FROM issue_dependency WHERE issue_id NOT IN (SELECT id FROM issue);
DELETE FROM issue_dependency WHERE dependency_id NOT IN (SELECT id FROM issue);
```
